### PR TITLE
Quit function implementation

### PR DESCRIPTION
**Pull request by Astrael1**

### CODE REVIEW

File: PolyEngine/Engine/Src/Engine.cpp [hunk 1]
```diff
@@ -106,3 +106,10 @@ void Engine::ResizeScreen(const ScreenSize & size)
 	gConsole.LogDebug("Screen resize: {} {}", size.Width, size.Height);
 	GetRenderingDevice()->Resize(size);
 }
+
+void Engine::QuitGame()
+{
+	shouldQuit = 1;
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Engine.cpp]: Please nevere use numbers for boolean value assignments. Use `true` and `false` instead. Also this implementation can be written in one line.

File: PolyEngine/Engine/Src/Engine.hpp [hunk 2]
```diff
@@ -123,6 +123,11 @@ namespace Poly
 		/// <param name="pos">New wheel position.</param>
 		void UpdateWheelPos(const Vector& pos) { InputEventsQueue.PushBack({eInputEventType::WHEELMOVE, pos}); }
 
+		//functions for closing the game
+		bool ShouldQuit();
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Engine.hpp]: `ShouldQuit` should be implemented as const method (`void ShouldQuit() const;`). Also please provide XML documentation for this method describing how is it used. I also suggest rename to `IsGameQuiteRequested`.

File: PolyEngine/Engine/Src/Engine.hpp [hunk 3]
```diff
@@ -123,6 +123,11 @@ namespace Poly
...
+		void QuitGame();
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Engine.hpp]: Let's rename it to `RequestGameQuit`. Also please provide XML documentation (simmilar to other methods in this file).

File: PolyEngine/Engine/Src/Engine.hpp [hunk 4]
```diff
@@ -223,6 +228,8 @@ namespace Poly
 
 		std::unordered_map<std::type_index, size_t> ComponentTypeMap;
 		std::unordered_map<std::type_index, size_t> WorldComponentTypeMap;
+
+		bool shouldQuit = 0; //stop the game
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Engine.hpp]: According to our coding style members should be names using PascalCase, not cameCase. Also let's rename it to `GameQuitRequested` and initialize with `false` and not 0.

File: PolyEngine/Standalone/Windows/Src/Main.cpp [hunk 5]
```diff
@@ -38,6 +38,8 @@ Poly::IGame* LoadGame()
 	return getGame();
 }
 
+
+
```
Comment (MuniuDev) [inline PolyEngine/Standalone/Windows/Src/Main.cpp]: Remove these unnecessary lines.

File: PolyEngine/Standalone/Windows/Src/Main.cpp [hunk 6]
```diff
@@ -218,3 +221,5 @@ LRESULT CALLBACK WindowProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lPara
 	// Handle any messages the switch statement didn't
 	return DefWindowProc(hWnd, message, wParam, lParam);
 }
+
+
```
Comment (MuniuDev) [inline PolyEngine/Standalone/Windows/Src/Main.cpp]: Remove these unnecessary lines.

File: PolyEngine/Standalone/Windows/Src/Main.cpp [hunk 7]
```diff
@@ -126,6 +128,7 @@ int WINAPI WinMain(HINSTANCE hInstance,
 				quitRequested = true;
 		}
 
+		quitRequested |= Engine.ShouldQuit();
```
Comment (MuniuDev) [inline PolyEngine/Standalone/Windows/Src/Main.cpp]: `|=` performs bitwise or and assign operation which is invalid in this context. Please replace it with `quitRequested = quitRequested || Engine.ShouldQuit();`. Also let's move this implementation after call to `Update`.

File: PolyEngine/Engine/Src/Engine.hpp (hunk 3 above)
Comment (VukoDrakkeinen) [reply PolyEngine/Engine/Src/Engine.hpp]: Hm, I don't know... `RequestGameQuit()`/`IsGameQuitRequested()` is grammatically incorrect and sounds like quitting is optional and can be denied.
How about `QuitGame()`/`GameAboutToQuit()` or `QuitGame()`/`IsGameQuitting()`?

Comment (VukoDrakkeinen) [reply PolyEngine/Engine/Src/Engine.hpp]: Apparently _quit_ as a noun is a collective term for some types of bird species, as I was surprised to find out 😛

File: PolyEngine/Standalone/Windows/Src/Main.cpp (hunk 7 above)
Comment (VukoDrakkeinen) [reply PolyEngine/Standalone/Windows/Src/Main.cpp]: One wonders why there is no corresponding `||=` operator, eh? 😉

**Pull request merged**